openerp/addons/base/res/res_partner.py

```diff
@@ -546,6 +571,21 @@ def create(self, vals):
         # migrating to the new API
         c_type = vals.get('company_type', self._context.get('default_company_type'))
         is_company = vals.get('is_company', self._context.get('default_is_company'))
+        parent_contact = self.env['res.partner'].search([('id', '=', vals.get('parent_id'))], limit=1)
+        gravatar = vals.get('email') and self._get_gravatar_image(vals['email']) or False
+        if parent_contact:
```

Comment:
All this code should be done only if image is not set. The order is user input > gravatar > default image (invoice / delivery / person / company).